crates/hyperqueue/src/server/event/log/write.rs

```diff
@@ -21,17 +31,21 @@ impl EventLogWriter {
         file.write_all(&header).await?;
         file.flush().await?;
 
-        let file = GzipEncoder::with_quality(file, Level::Fastest);
-
-        // Keep buffer capacity larger than max size to avoid reallocation if we overflow
-        // the buffer.
-        let buffer = Vec::with_capacity(BUF_MAX_SIZE * 2);
-        Ok(Self { file, buffer })
+        let buffer = Vec::with_capacity(CHUNK_MAX_SIZE * 2);
+        Ok(Self {
+            file,
+            buffer,
+            first_event_in_buffer: 0,
+        })
     }
 
     #[inline]
     pub async fn store(&mut self, event: MonitoringEvent) -> anyhow::Result<()> {
+        if self.buffer.is_empty() {
```

Comment:
What happens if we flush/finish the writer without storing a single event? Then the chunk header would say that the first event ID is 0, which is wrong. We should model this as an enum, something like 
```rust
enum WriterState {
    EmptyChunk(Vec<u8>),
    ChunkInProgress {
        first_event_id: MonitoringEventId,
        buffer: Vec<u8>
    }
}
```
and define corresponding state transition methods between these two variants. Storing an event while empty chunk => go to chunk in progress. Flushing => if not empty chunk, go from chunk in progress to empty chunk.

The only reason to include the `Vec<u8>` in the `EmptyChunk` variant is so that we can reuse its allocated buffer. Otherwise we would need to recreate it each time we go from `EmptyChunk` to `ChunkInProgress`.